testing/framework/TestCmdTests.py

```diff
@@ -1503,12 +1503,11 @@ def cleanup_test(test, cond=None, stdout=""):
             test.preserve('pass_test')
             cleanup_test(test, 'pass_test', "Preserved directory %s\n" % wdir)
             assert os.path.isdir(wdir)
-            cleanup_test(test, 'fail_test')
+            shutil.rmtree(wdir, ignore_errors = 1)
```

Comment:
Why this change?

Reply:
The tests were relying on behavior where cleanup_test can be called many times with a 'pass_test' result. Then one you call it with a 'fail_test' result, it will remove all directories. AFAICT this is not normal behavior; a test will either pass or succeed and we should clean up once.

Reply:
The issue with this change is that it's not respecting the PRESERVE, PRESERVE_FAIL shell environment variables not deleting test directories when set..